Updated various pages to use the langchain-neo4j package (#343)
neo4j-documentation/developer-guides: modules/genai-ecosystem/pages/langchain.adoc

The broad and deep Neo4j integration allows for vector search, cypher generation and database querying and knowledge graph construction.

> Here is an overview of the https://python.langchain.com/docs/tutorials/graph/[Graph Integrations^].

Reviewer: ```suggestion
Here is an overview of the link:https://python.langchain.com/docs/tutorials/graph/[Graph Integrations^].
```

> * https://towardsdatascience.com/integrating-neo4j-into-the-langchain-ecosystem-df0e988344d2[Integrating Neo4j into the LangChain Ecosystem^]

Reviewer: ```suggestion
* link:https://towardsdatascience.com/integrating-neo4j-into-the-langchain-ecosystem-df0e988344d2[Integrating Neo4j into the LangChain Ecosystem^]
```

> * https://blog.langchain.dev/implementing-advanced-retrieval-rag-strategies-with-neo4j/[Implementing Advanced Retrieval RAG Strategies with Neo4j^]

Reviewer: ```suggestion
* link:https://blog.langchain.dev/implementing-advanced-retrieval-rag-strategies-with-neo4j/[Implementing Advanced Retrieval RAG Strategies with Neo4j^]
```

Creating a Knowledge Graph from unstructured data like PDF documents used to be a complex and time-consuming task that required training and using dedicated, large NLP models.

> The https://python.langchain.com/v0.1/docs/use_cases/graph/constructing[Graph Transformers^] are tools that allows you to extract structured data from unstructured documents and transform it into a Knowledge Graph.

Reviewer 2: The https://python.langchain.com/v0.1/docs/use_cases/graph/constructing[Graph Transformers^] are tools that extract structured data from unstructured documents and transform it into a Knowledge Graph.

Author: Updated in [36a8f7e](https://github.com/neo4j-documentation/developer-guides/pull/343/commits/36a8f7e369f0b96a2e8c651384e0702742e8af28)

> The https://python.langchain.com/v0.1/docs/use_cases/graph/constructing[Graph Transformers^] are tools that allows you to extract structured data from unstructured documents and transform it into a Knowledge Graph.

Reviewer: ```suggestion
The link:https://python.langchain.com/v0.1/docs/use_cases/graph/constructing[Graph Transformers^] are tools that allow you to extract structured data from unstructured documents and transform it into a Knowledge Graph.
```

> NOTE: You can see a practical application, code and demo for extracting knowledge graphs from PDFs, YouTube transcripts, wikpedia articles and more with the xref:llm-graph-builder.adoc[LLM Graph Builder].

Reviewer: ```suggestion
[NOTE]
====
You can see a practical application, code and demo for extracting knowledge graphs from PDFs, YouTube transcripts, wikpedia articles and more with the xref:llm-graph-builder.adoc[LLM Graph Builder].
====
```

image::

* https://python.langchain.com/docs/integrations/graphs/diffbot/[Diffbot Graph Transformer^]

> * https://python.langchain.com/v0.1/docs/use_cases/graph/constructing/#llm-graph-transformer[LLM Graph Transformer^]

Reviewer: ```suggestion
* link:https://python.langchain.com/docs/integrations/graphs/diffbot/[Diffbot Graph Transformer^]
* link:https://python.langchain.com/v0.1/docs/use_cases/graph/constructing/#llm-graph-transformer[LLM Graph Transformer^]
```

Reviewer: And this format everywhere else you're linking outside.

== Documentation

* https://python.langchain.com/docs/integrations/providers/neo4j/[Neo4j Integrations^]
* https://python.langchain.com/docs/tutorials/graph/#chain[Graph Cypher QA Chain^]
* https://python.langchain.com/docs/integrations/vectorstores/neo4jvector[Neo4j Vector^]
* https://python.langchain.com/v0.1/docs/use_cases/graph/constructing[Graph Transformers^]

== Starter Kit

> This https://github.com/neo4j-examples/langchain-starter-kit/blob/main/app/vector_chain.py[starter-kit] demonstrates how to run a FastAPI server using LangChain to answer queries on data stored in a Neo4j instance. The single endpoint can be used to retrieve answers using either a https://python.langchain.com/docs/integrations/vectorstores/neo4jvector/[Vector index chain],  https://python.langchain.com/docs/integrations/graphs/neo4j_cypher/[GraphCypherQA Chain], or a composite answer of both. The https://github.com/neo4j-examples/langchain-starter-kit/tree/langserve[langserve] branch contains an example of the same service, using https://python.langchain.com/v0.1/docs/langserve/[LangServe]

Reviewer: ```suggestion
This link:https://github.com/neo4j-examples/langchain-starter-kit/blob/main/app/vector_chain.py[starter-kit] demonstrates how to run a FastAPI server using LangChain to answer queries on data stored in a Neo4j instance. 
The single endpoint can be used to retrieve answers using either a link:https://python.langchain.com/docs/integrations/vectorstores/neo4jvector/[Vector index chain],  link:https://python.langchain.com/docs/integrations/graphs/neo4j_cypher/[GraphCypherQA Chain], or a composite answer of both. 
The link:https://github.com/neo4j-examples/langchain-starter-kit/tree/langserve[langserve] branch contains an example of the same service, using link:https://python.langchain.com/v0.1/docs/langserve/[LangServe].
```
